ui/src/app/applications/components/application-operation-state/application-operation-state.tsx

```diff
@@ -54,11 +54,59 @@ const Filter = (props: {filters: string[]; setFilters: (f: string[]) => void; op
 
 export const ApplicationOperationState: React.StatelessComponent<Props> = ({application, operationState}, ctx: AppContext) => {
     const [messageFilters, setMessageFilters] = React.useState([]);
+    const getFormattedMessage = (message: string) => {
+        const prefix = 'one or more objects failed to apply, reason: ';
+        let cleanMessage = message;
+
+        // Remove duplicate prefix if exists
+        if (message.startsWith(prefix) && message.substring(prefix.length).startsWith(prefix)) {
+            cleanMessage = prefix + message.substring(prefix.length * 2);
+        }
+
+        // Format immutable fields error message
+        if (cleanMessage.includes('attempting to change immutable fields:')) {
```

Comment:
maybe it will be better to move this part to some formatter util that will cover such cases and use it here? i am sure we probably will want cover other formatting cases in future